components/automate-ui/src/app/pages/data-feed/data-feed.component.ts

```diff
@@ -0,0 +1,136 @@
+    private service: DatafeedService,
+    public dialog: MatDialog,
+    public snackBar: MatSnackBar,
+    private telemetryService: TelemetryService
+  ) { }
+
+  ngOnInit() {
+    this.destinations$ = this.service.fetchDestinations();
+    this.destinations$.subscribe(destinations => {
+        this.sendCountToTelemetry(destinations);
+      },
+      error => {
+        if (error.status === 403) {
+          this.permissionDenied = true;
+        } else  {
+          this.errorLoading = true;
+        }
+      }
+    );
+    this.resetSortDir();
+    this.toggleSort('name');
+  }
+
+  toggleSort(field: string) {
+    if (field === this.sortField) {
+      // when sorting is inverted for the currently sorted column
+      this.sortDir[field] = this.sortDir[field] === 'asc' ? 'desc' : 'asc';
+    } else {
+      // when sorting a different column than the currently sorted one
+      this.resetSortDir();
+    }
+    this.sortField = field;
+    this.updateSort(field, this.sortDir[field]);
+  }
+  sortIcon(field: string): string {
+    if (field === this.sortField) {
+      return 'sort-' + this.sortDir[field];
+    } else {
+      return 'sort-asc';
+    }
+  }
+
+  deleteDestination(destination: Destination) {
+    const dialogRef2 = this.dialog.open(DeleteDatafeedDialogComponent);
+    dialogRef2.afterClosed().pipe(
+      filter((result: any) => result === 'delete' ))
+      .subscribe(_result => {
+        this.service.deleteDestination(destination).subscribe(_res => {
+          this.snackBarMessage(`Destination '${destination.name}' was deleted.`);
+          this.refreshDestinations();
+        }, err => {
+          const body = err;
+          this.snackBarMessage(`Could not delete destination '${destination.name}': ${body}`);
+        });
+      });
+  }
+
+  snackBarMessage(message) {
+    this.snackBar.open(message, '', { duration: 6000 } );
+  }
+
+  private updateSort(field: string, direction: string) {
+    this.destinations$ = this.destinations$.pipe(map(destinations => {
+      let sortedDestinations: Destination[] = [];
+      if (field === 'name') {
+        sortedDestinations = destinations.sort((r1: Destination, r2: Destination) => {
+          return r1.name.localeCompare(r2.name);
+        });
+      }
+      if (direction === 'asc') {
+        return sortedDestinations;
+      } else {
+        return sortedDestinations.reverse();
+      }
+    }));
+  }
+
+  private resetSortDir(): void {
+    this.sortDir = {
+      name: 'asc',
+      url: 'asc'
+    };
+  }
+
+  private sendCountToTelemetry(destinations: Destination[]) {
+    this.telemetryService.track('datafeedDestinationsCount', destinations.length);
+  }
+
+
+  // TODO - this was common in all three uses, but I'm not sure this is the best
+  // way to do it - do we really need to refresh after the server confirms the action
+  // successful? Seem it should be possible to update the local model with the
+  // changes and have that trigger view updates?
```

Comment:
I would get rid of this comment. It just shows this code was copied from the NotificationsComponent 😄.